### DISCUSSION TITLE
How to Properly Handle "attempt to index field 'inlay_hints' (a nil value)" Error in LazyVim LSP Config?

### DISCUSSION THREAD

**Discussion by anewplz** [LazyVim/LazyVim#5273] [Q&A] [answered]
I'm using LazyVim with nvim-lspconfig and recently encountered an issue after trying to customize my LSP keymaps and modify the K key behavior. Here's the situation:
I wanted to remap the default hover functionality (K) for a different purpose (5k). Here's the configuration I used:
`return{
   "neovim/nvim-lspconfig",
    opts = function()
    local keys = require("lazyvim.plugins.lsp.keymaps").get()
    
    keys[#keys + 1] = { "K", false }

    keys[#keys + 1] = { "K", "5k", desc = "Move 5 lines up" }

    return keys  
  end,
}`
After applying this configuration, I got the following error:
`Failed to run `config` for nvim-lspconfig
...share/nvim/lazy/LazyVim/lua/lazyvim/plugins/lsp/init.lua:141: attempt to index field 'inlay_hints' (a nil value)
`
The stack trace points to the inlay_hints functionality in LazyVim's LSP configuration, specifically this block in init.lua:
`--inlay hinnlay hints
if opts.inlay_hints.enabled then
  LazyVim.lsp.on_supports_method("textDocument/inlayHint", function(client, buffer)
    if
      vim.api.nvim_buf_is_valid(buffer)
      and vim.bo[buffer].buftype == ""
      and not vim.tbl_contains(opts.inlay_hints.exclude, vim.bo[buffer].filetype)
    then
      vim.lsp.inlay_hint.enable(true, { bufnr = buffer })
    end
  end)
endts
if opts.inlay_hints.enabled then
  LazyVim.lsp.on_supports_method("textDocument/inlayHint", function(client, buffer)
    if
      vim.api.nvim_buf_is_valid(buffer)
      and vim.bo[buffer].buftype == ""
      and not vim.tbl_contains(opts.inlay_hints.exclude, vim.bo[buffer].filetype)
    then
      vim.lsp.inlay_hint.enable(true, { bufnr = buffer })
    end
  end)
end`

I resolved the error by disabling inlay_hints in my configuration (opts.inlay_hints.enabled = false).
But i still want to use t inlay hints,I'm new to neovim and lua ,is there any better solutions?
Any help or insights would be greatly appreciated! Thank you! 🙏

**Comment by dpetka2001** [comment]
What is your Neovim version?

**Reply by dpetka2001** [reply, marked as answer]
Remove the `return keys` from your `opts` function. That's what's causing the prob.

**Reply by anewplz** [reply]
version is 0.10.3，and your answer works,thanks a lot.Wishing you all the best！

**Reply by MundaneImmortal** [reply]
The same problem and the answer works for me too. Thank you guys!
